core/src/env/traits.rs

```diff
@@ -127,4 +130,22 @@ pub trait Env: EnvTypes {
 
     /// Deposits raw event data through Contracts module.
     fn deposit_raw_event(topics: &[<Self as EnvTypes>::Hash], data: &[u8]);
+
+
+    /// Calls a remote smart contract and return access to scratch buffer without return data
+    fn call_invoke(
+        callee: <Self as EnvTypes>::AccountId,
+        gas: u64,
+        value: <Self as EnvTypes>::Balance,
+        input_data: &[u8]
+    );
```

Comment:
For `call_invoke` we want to return `Result<(), CallError>` since the call to another smart contract might be errorneous. `CallError` is simply a `struct CallError;` that we can extend later on.